src/schemas/tokens-info.ts

```diff
@@ -26,34 +26,61 @@ const checkMinVersion: CustomValidator = (value) => {
   }
 }
 
-const TokensInfoSchema = Joi.object().pattern(
-  AddressSchema,
+const BaseTokenInfoSchema = Joi.object({
+  address: AddressSchema,
+  imageUrl: Joi.string()
+    // For now only allow assets within this repo
+    .pattern(
+      /^https:\/\/raw.githubusercontent.com\/valora-inc\/address-metadata\/main\/assets\/tokens\/[^/]+\.png$/,
+    )
+    .uri()
+    .custom(checkMatchingAsset, 'has a matching asset'),
+  name: Joi.string().required(),
+  decimals: Joi.number().required(),
+  symbol: Joi.string().required(),
+  isCoreToken: Joi.boolean(),
+  isSupercharged: Joi.boolean(),
+  pegTo: AddressSchema,
+  // Forcing flag to be false if decimals is not 18
+  isSwappable: Joi.when('decimals', {
+    is: 18,
+    then: Joi.boolean(),
+    otherwise: Joi.valid(false),
+  }),
+  minimumAppVersionToSwap: Joi.string()
+    .pattern(/^\d+\.\d+\.\d+$/)
+    .custom(checkMinVersion, 'has a valid version'),
+  isNative: Joi.boolean(),
+})
+
+export const TokenInfoSchema = Joi.alternatives().try(
+  Joi.object({
+    // native tokens don't have an address except CELO
+    isNative: Joi.valid(true).required(),
+    symbol: Joi.string().invalid('CELO').required(),
+    address: Joi.forbidden(),
+  }).concat(BaseTokenInfoSchema),
   Joi.object({
+    // CELO is native and has an address
+    isNative: Joi.valid(true).required(),
+    symbol: Joi.valid('CELO').required(),
```

Comment:
Maybe use a list here?

Reply:
like `Joi.valid(...NATIVE_TOKENS_WITH_ADDRESSES)`? I mentioned this in another comment-- let's do it when we need it.